modules/core/src/test/java/org/apache/ignite/cdc/ChangeDataCaptureOrderTest.java

```diff
@@ -0,0 +1,250 @@
+    }
+
+    /** */
+    @Test
+    public void testOrderIncrease() throws Exception {
+        IgniteConfiguration cfg = getConfiguration("ignite-0");
+
+        IgniteEx ign = startGrid(cfg);
+
+        ign.cluster().state(ACTIVE);
+
+        AtomicLong updCntr = new AtomicLong(0);
+        int key = 42;
+
+        ChangeDataCaptureConsumer cnsmr = new ChangeDataCaptureConsumer() {
+            private long order = -1;
+
+            @Override public boolean onEvents(Iterator<ChangeDataCaptureEvent> evts) {
+                evts.forEachRemaining(evt -> {
+                    assertEquals(key, evt.key());
+
+                    assertTrue(evt.version().order() > order);
+
+                    order = evt.version().order();
+
+                    updCntr.incrementAndGet();
+                });
+
+                return true;
+            }
+
+            @Override public void start() {
+                // No-op.
+            }
+
+            @Override public void stop() {
+                // No-op.
+            }
+        };
+
+        ChangeDataCapture cdc = new ChangeDataCapture(cfg, null, cdcConfig(cnsmr));
+
+        IgniteCache<Integer, User> cache = ign.getOrCreateCache("my-cache");
+
+        IgniteInternalFuture<?> fut = runAsync(cdc);
+
+        for (int i = 0; i < KEYS_CNT; i++)
+            cache.put(key, new User("John Connor " + i, 42 + i, null));
+
+        assertTrue(waitForCondition(() -> updCntr.get() == KEYS_CNT, getTestTimeout()));
+
+        fut.cancel();
+    }
+
+    /** */
+    private void addConflictData(IgniteCache<Integer, User> cache, int from, int to) {
+        try {
+            IgniteEx ign = (IgniteEx)G.allGrids().get(0);
+
+            IgniteInternalCache intCache = ign.cachex(cache.getName());
+
+            Map<KeyCacheObject, GridCacheDrInfo> drMap = new HashMap<>();
+
+            for (int i = from; i < to; i++) {
+                byte[] bytes = new byte[1024];
+
+                ThreadLocalRandom.current().nextBytes(bytes);
+
+                KeyCacheObject key = new KeyCacheObjectImpl(i, null, intCache.affinity().partition(i));
+                CacheObject val =
+                    new CacheObjectImpl(new User("John Connor " + i, 42 + i, bytes), null);
+
+                val.prepareMarshal(intCache.context().cacheObjectContext());
+
+                drMap.put(key, new GridCacheDrInfo(val, new GridCacheVersion(1, i, 1, OTHER_DR_ID)));
+            }
+
+            intCache.putAllConflict(drMap);
+        }
+        catch (IgniteCheckedException e) {
+            throw new IgniteException(e);
+        }
+    }
+
+    /** */
+    public static class TestCacheConflictResolutionManager<K, V> extends GridCacheManagerAdapter<K, V>
+        implements CacheConflictResolutionManager<K, V> {
+        @Override public CacheVersionConflictResolver conflictResolver() {
+            return new CacheVersionConflictResolver() {
+                @Override public <K, V> GridCacheVersionConflictContext<K, V> resolve(
```

Comment:
seems you have broken generics here.
These aren't the K,V you're looking for :)